clang/test/SemaSYCL/names_in_lambda_nullptr.cpp

```diff
@@ -0,0 +1,24 @@
+// RUN: %clang_cc1 -fsycl-is-device -std=c++20 -fsyntax-only %s -verify
+// RUN: %clang_cc1 -fsycl-is-host -std=c++20 -fsyntax-only %s -verify
+
+// This test checks that when a binding declaration is captured that
+// we don't dereference the null VarDecl incorrectly.
+
+// CHECK: expected-no-diagnostics
+
+template <typename T>
```

Comment:
Can the test be simpler? Something like the following should be enough to test this right?

```
int main() {
  int a[2] = {1, 2};
  auto [x, y] = a;
  auto L = [=]() {x = 10;};
  L();
  return 0;
}
```

Reply:
Also can these kind of captures happen inside SYCL kernel and if so should we add a test to see what metadata is generated?

Reply:
> Can the test be simpler? Something like the following should be enough to test this right?
> 
> ```
> int main() {
>   int a[2] = {1, 2};
>   auto [x, y] = a;
>   auto L = [=]() {x = 10;};
>   L();
>   return 0;
> }
> ```

Done.

Reply:
> Also can these kind of captures happen inside SYCL kernel and if so should we add a test to see what metadata is generated?

I have added a CodeGen test.